include/led.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2018 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _ZEPHYR_LED_H
+#define _ZEPHYR_LED_H
+
+/**
+ * @file
+ * @brief Public LED driver APIs
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
+
+/**
+ * @brief Internal driver specific representation of an LED device
+ */
+
+struct led_data {
+	/* Minimum acceptable LED blinking time period (in ms) */
+	u32_t min_period;
+	/* Maximum acceptable LED blinking time period (in ms) */
+	u32_t max_period;
+	/* Minimum acceptable LED brightness value */
+	u8_t min_brightness;
+	/* Maximum acceptable LED brightness value */
+	u8_t max_brightness;
+};
+
+/**
+ * @typedef led_api_blink()
+ * @brief Callback API for blinking an LED
+ *
+ * @see led_blink() for argument descriptions.
+ */
+typedef int (*led_api_blink)(struct device *dev, u32_t led, u32_t period);
+
+/**
+ * @typedef led_api_set_brightness()
+ * @brief Callback API for setting brightness of an LED
+ *
+ * @see led_set_brightness() for argument descriptions.
+ */
+typedef int (*led_api_set_brightness)(struct device *dev, u32_t led,
+				      u8_t value);
+/**
+ * @typedef led_api_on()
+ * @brief Callback API for turning on an LED
+ *
+ * @see led_on() for argument descriptions.
+ */
+typedef int (*led_api_on)(struct device *dev, u32_t led);
+
+/**
+ * @typedef led_api_off()
+ * @brief Callback API for turning off an LED
+ *
+ * @see led_off() for argument descriptions.
+ */
+typedef int (*led_api_off)(struct device *dev, u32_t led);
+
+/**
+ * @brief LED driver API
+ *
+ * This is the mandatory API any LED driver needs to expose.
+ */
+struct led_driver_api {
+	led_api_blink blink;
+	led_api_set_brightness set_brightness;
+	led_api_on on;
+	led_api_off off;
+};
+
+/**
+ * @brief Blink an LED
+ *
+ * This routine blinks an LED for the given time period continuously
```

Comment:
What duty cycle will it blink at?

Reply:
That depends on the value set by set_brightness. Duty cycle or brightness has to be set before calling this function. Should we need to mention it explicitly?

Reply:
Ah, sorry, by 'duty cycle' I mean how much of the cycle will the LED be on for.  There's a few examples at https://en.wikipedia.org/wiki/Light_characteristic including:

- Flashing (50 % on, 50 % off)
- Occulting (~20 % on, 80 % off)
- Reverse occulting (~80 % on, 20 % off)

'Flashing' SGTM.

Reply:
Not sure which category makes sense here. For this `led_blink` function, the device is expected to be ON for given period and OFF for the same period. For instance, if the period is 500ms then first 500ms it will be ON and next 500ms it will be OFF.